### PR TITLE
feat:added query stream() method and removed fetchAll() 

### PR DESCRIPTION
**Pull request by medgar-nitric**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,7 +11,8 @@
 
 # Nitric Java SDK
 The Java SDK supports the use of the cloud-portable [Nitric](https://nitric.io) framework with Java 11.
-> The SDK is in early stage development and APIs and interfaces are still subject to breaking changes
+
+> The SDK is available as early access and interfaces may still subject to breaking changes.
```
Comment (tjholm) [inline README.md:15]: "...interfaces  ["are" | "are still" | "may still be"] subject to breaking changes".

File: src/main/java/io/nitric/api/document/Query.java [hunk 2]
```diff
@@ -81,7 +78,7 @@
  *  import io.nitric.api.document.QueryResults;
  *  import com.example.entity.Customer;
  *
- *  QueryResults results = Documents.collection("customers")
+ *  QueryResults&lt;Customer&gt; results = Documents.collection("customers")
```
Comment (tjholm) [inline src/main/java/io/nitric/api/document/Query.java:81]: Not change required but worth discussing later: Using markdown for code examples eliminates the need to remember to use character encoding, making it easier to read/maintain.

**Review by tjholm — CHANGES_REQUESTED** [review]

File: src/main/java/io/nitric/api/document/QueryResults.java [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+                    DocumentQueryResponse response = null;
+                    try {
+                        response = Documents.getServiceStub().query(request);
+                    } catch (io.grpc.StatusRuntimeException sre) {
+                        throw ProtoUtils.mapGrpcError(sre);
+                    }
+
+                    queryResults.loadPageData(response);
+
+                    return queryResults.queryData.size() > 0;
```
Comment (tjholm) [inline src/main/java/io/nitric/api/document/QueryResults.java:180]: `return !queryResults.queryData.isEmpty()`;

**Review by tjholm — CHANGES_REQUESTED** [review]

File: src/main/java/io/nitric/api/document/QueryResults.java (hunk 3 above)
Comment (medgar-nitric) [reply src/main/java/io/nitric/api/document/QueryResults.java]: Done

File: src/main/java/io/nitric/api/document/Query.java (hunk 2 above)
Comment (medgar-nitric) [reply src/main/java/io/nitric/api/document/Query.java:81]: Agreed

File: README.md (hunk 1 above)
Comment (medgar-nitric) [reply README.md]: Done

**Review by tjholm — APPROVED** [review]

**Review by HomelessDinosaur — APPROVED** [review]

**Pull request merged**